zerver/lib/send_email.py

```diff
@@ -148,8 +150,15 @@ def render_templates() -> Tuple[str, str, str]:
     if from_address == FromAddress.support_placeholder:
         from_address = FromAddress.SUPPORT
 
-    # Set the "From" that is displayed separately from the envelope-from
-    extra_headers["From"] = str(Address(display_name=from_name, addr_spec=from_address))
+    # Set the "From" that is displayed separately from the envelope-from.
+    # Check that the sender address is not overly-long for SES compatibility.
+    # If the address is still too long without the name an SMTPDataError will
+    # be raised when sending the email and it will be caught to raise an
+    # EmailNotDeliveredException
+    if getsizeof(from_name + from_address) > 320:
```

Comment:
Is this expression exactly the thing we should be checking the size of, based on the RFC? Or should we be checking the properly formatted string, so something of the form "name <address@example.com>", which would be slightly longer

Reply:
The RFC's phrasing is describing the bytes of the address itself as being limteid to 320, and doesn't include the name:

> In addition to restrictions on syntax, there is a length limit on email addresses.  That limit is a maximum of 64 characters (octets) in the "local part" (before the "@") and a maximum of 255 characters (octets) in the domain part (after the "@") for a total length of 320 characters.

It's quite possible that SES is applying the limit in the wrong place (to the fully-formatted string, not the address part).   

But we should compare the length of what we actually generated, not a best-estimate:
```
extra_headers["From"] = str(Address(display_name=from_name, addr_spec=from_address))
if len(extra_headers["From"]) > 320:
    extra_headers["From"] = str(Address(addr_spec=from_address))
```